code/clue.js

```diff
@@ -43,22 +157,71 @@ const randomSelector = array => {
 // With a killer, a weapon and a room.
 // The values will be set later.
 
+const mystery = {}
+
 // FINISH THIS FUNCTION TO SHOW ALL INFORMATION ABOUT THE KILLER.
 // This function will be invoked when you click on the killer card.
 const pickKiller = () => {
   // This will randomly select a killer from the suspects. And add that to the mystery object.
   mystery.killer = randomSelector(suspects)
 
+  console.log(pickKiller);
+
   // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer.
   const theKiller = document.getElementById("killer")
   const theKillerName = document.getElementById("killerName")
+  const theKillerAge = document.getElementById("killerAge")
+  const theKillerOccupation = document.getElementById("killerOccupation")
+  const theKillerDescription = document.getElementById("killerDescription")
+  const theKillerImage = document.getElementById("killerImage")
+
 
   theKiller.style.background = mystery.killer.color
   theKillerName.innerHTML =
     mystery.killer.firstName + " " + mystery.killer.lastName
+  theKillerAge.innerHTML =
+    mystery.killer.age
+  theKillerOccupation.innerHTML =
+    mystery.killer.occupation
+  theKillerDescription.innerHTML =
+    mystery.killer.description
+  theKillerImage.setAttribute("src", mystery.killer.image)
 }
 
 // CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
 
+const pickWeapon = () => {
+  mystery.weapon = randomSelector(weapons)
+
+  const theWeapon = document.getElementById("weaponName")
+
+  theWeapon.innerHTML =
+    mystery.weapon.name
+
+}
+
+console.log(pickWeapon);
+
+const pickRoom = () => {
+  mystery.room = randomSelector(rooms)
+
+  const theRoom = document.getElementById("roomName")
+
+  theRoom.innerHTML =
+    mystery.room
+
+}
+
+console.log(pickRoom);
+
 // CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
 // 'The murder was committed by Jacob Green, in the living room with a rope.'
+
+const revealMystery = () => {
+
+  const theMystery = document.getElementById("mystery")
+
+  theMystery.innerHTML =
+    (`The murder was committed by ${mystery.killer.firstName} ${mystery.killer.lastName}, in the ${mystery.room} with a ${mystery.weapon.name}.`)
```

Comment:
Here you got an error when you are clicking on the button, but hasn't chosen a room, killer and weapon. For handling this you can do an if statement here. For example you can check if `mystery.killer.firstName` exists